### PR TITLE
Added spicedb client to repository, generated config and added wire c…

### PR DESCRIPTION
**Pull request by merlante**
Integrate spiceDB client in the Relations API service code

## Describe your changes

- SpiceDbRepository added as impl of ZanzibarRepository.
- A relationships create method that may be working. (Outside the scope of this PR to test functionally.)
- New config.proto and generated files to remove database/redis and add spicedb config.
- wire.go config to wire up and run everything.

## Ticket reference (if applicable)
Fixes #RHCLOUD-30386

## Checklist

* [x] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [x] Do your changes have passing automated tests and sufficient observability?

* [x] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [x] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [x] Are the agreed upon coding/architectural practices applied?

* [x] Are security needs fullfilled? (e.g. no internal URL)

* [x] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [x] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

**Comment by Rajagopalan-Ranganathan** [comment]
@merlante can you mark this as "WIP" , thanks!

File: cmd/ciam-rebac/wire.go [hunk 1]
```diff
@@ -17,5 +19,5 @@ import (
 
 // wireApp init kratos application.
 func wireApp(*conf.Server, *conf.Data, log.Logger) (*kratos.App, func(), error) {
-	panic(wire.Build(server.ProviderSet, service.ProviderSet, newApp))
+	panic(wire.Build(server.ProviderSet, service.ProviderSet, biz.ProviderSet, data.ProviderSet, newApp))
```
Comment (akoserwal) [inline cmd/ciam-rebac/wire.go:22]: You don't have to make change here, 

you can refer: https://github.com/RedHatInsights/insights-rebac/blob/main/internal/service/service.go
for adding provider

Comment (merlante) [reply cmd/ciam-rebac/wire.go:22]: Is that the best place? I thought the pattern was to add a ProviderSet at each layer, and then add all of the sets in wireApp with wire.Build. (But I'm new to wire.go, so I could be wrong.)

Comment (akoserwal) [reply cmd/ciam-rebac/wire.go:22]: Yes, that's what I mean by adding a provider, adding an ad for each layer. I think did correctly here, got confused.

**Review by merlante — COMMENTED** [review]
Should be there or thereabouts now.

**Review by akoserwal — APPROVED** [review]

**Pull request merged**